dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py

```diff
@@ -0,0 +1,276 @@
+        processes = self.get_json_response(endpoint)
+        return processes
+
+    def collect_hosts(self):
+        """
+        Collects the response from the Dynatrace Host API endpoint
+        """
+        endpoint = self.url + "/api/v1/entity/infrastructure/hosts"
+        hosts = self.get_json_response(endpoint)
+        return hosts
+
+    def collect_applications(self):
+        """
+        Collects the response from the Dynatrace Application API endpoint
+        """
+        endpoint = self.url + "/api/v1/entity/applications"
+        applications = self.get_json_response(endpoint)
+        return applications
+
+    def collect_process_groups(self):
+        """
+        Collects the response from the Dynatrace Process-Group API endpoint
+        """
+        endpoint = self.url + "/api/v1/entity/infrastructure/process-groups"
+        process_groups = self.get_json_response(endpoint)
+        return process_groups
+
+    def collect_services(self):
+        """
+        Collects the response from the Dynatrace Service API endpoint
+        """
+        endpoint = self.url + "/api/v1/entity/services"
+        services = self.get_json_response(endpoint)
+        return services
+
+    def process_component(self, response, component_type):
+        """
+        Process each component type and map those with specific data
+        :param response: Response of each component type endpoint
+        :param component_type: Component type
+        :return: create the component on stackstate API
+        """
+        urn = ''
+        if "error" not in response:
+            for item in response:
+                # special case for host type as we get some float values
+                if component_type == "host":
+                    for key in item.keys():
+                        if type(item[key]) is float:
+                            item[key] = str(item[key])
+                identifiers = []
+                data = dict()
+                externalId = item.get("entityId")
+                if component_type == "service":
+                    urn = "urn:service:/{}".format(externalId)
+                elif component_type == "process-group":
+                    urn = "urn:process-group:/{}".format(externalId)
+                elif component_type == "application":
+                    urn = "urn:application:/{}".format(externalId)
+                elif component_type == "process":
+                    urn = "urn:process:/{}".format(externalId)
+                elif component_type == "host":
+                    displayName = item.get("displayName")
+                    urn = "urn:host:/{}".format(displayName)
+                identifiers.append(urn)
+                # derive useful labels and get labels from dynatrace tags
+                labels = self.get_labels(item)
+                labels += self.get_labels_from_dynatrace_tags(item)
+                if "tags" in item:
+                    del item["tags"]
```

Comment:
Perhaps you should place this code in `filter_data`, then call it sooner and call it `filter_item_topology_data`.